### PR TITLE
Add a dtype format in the b2nd metalayer

### PR DESCRIPTION
**Pull request by FrancescAlted**
This allows other formats than NumPy for representing the dtype in the future.

### CODE REVIEW

File: README_B2ND_METALAYER.rst [hunk 1]
```diff
@@ -52,12 +52,16 @@ Next, the `blockshape` section is meant to store the actual block shape info::
...
+
+    |---|---|--4 bytes---|--------------|
+    | XX| db| dtype_len  | dtype_string |
+    |---|---|------------|--------------|
+      ^   ^
+      |   |
+      |   +--[msgpack] str32
+      +--[msgpack] positive fixint (7-bit integer). dtype_format; 0 means NumPy format.
+
+The 0 value for dtype_format means that that the dtype_string field follows the NumPy convention
```
Comment (martaiborra) [inline README_B2ND_METALAYER.rst:65]: "that that" -> "that"

**Review by martaiborra — COMMENTED** [review]
LGTM

File: README_B2ND_METALAYER.rst (hunk 1 above)
Comment (FrancescAlted) [reply README_B2ND_METALAYER.rst]: Well spotted.  Fixed in ee660bcb (+ a problem with a malloc size)

**Comment by FrancescAlted** [comment]
This should be ready for merging, except that MinGW platform consistently crashes (and GCC Ninja occasionally too).  @martaiborra is in charge to have a look at this.

**Pull request merged**